app/components/auth/firebase/ui/LoginForm.js

```diff
@@ -0,0 +1,162 @@
+import { 
+    EmailAuthProvider, 
+    FacebookAuthProvider, 
+    fetchSignInMethodsForEmail, 
+    getAuth, 
+    GoogleAuthProvider, 
+    linkWithCredential, 
+    OAuthProvider, 
+    signInWithEmailAndPassword, 
+    signInWithPopup 
+} from "firebase/auth";
+import {getApp} from 'firebase/app';
+import { useState } from "react";
+import { FormControl, Alert, Button } from "react-bootstrap";
+import {FB_APP_NAME} from '../lib/constants';
+export function LoginForm({onSignupClick}){
+    const [email,setEmail] = useState("");
+    const [password,setPassword] = useState("");
+    const [errorMessage,setError] = useState("");
+    const [diffCredError,setDiffCredError] = useState(null);
+    const [progress,setProgress] = useState(false);
+
+    const doEmailPasswordLogin = async (e) => {
+        e.preventDefault();
+        if(progress){
+            return true;
+        }
+        setProgress(true);
+        try {
+            const fbApp = getApp(FB_APP_NAME);
+            const userCred = await signInWithEmailAndPassword(getAuth(fbApp),email,password);
+
+            if(diffCredError?.oldProvider?.providerId === EmailAuthProvider.PROVIDER_ID){
+                // The signin was requested to link new credentials with the account 
+                await linkWithCredential(userCred.user,OAuthProvider.credentialFromError(diffCredError.error));
+            }
+
+        } catch(error){
+            switch(error.code){
```

Comment:
Check if `error` includes a `message` field. Directily `setError` with `error.message` instead of using switch-case.

Reply:
Firebase functions throw errors with a certain code. I check the error.message property. Its same as the code. In order to use custom error messages. I used switch cases.